File: 999.Available-Captures-for-Rook.py
# https://leetcode.com/problems/available-captures-for-rook/
# Easy (76.3%)
# Total Accepted:    2,857
# Total Submissions: 3,744
# beats 100.0% of python submissions


class Solution(object):
    def numRookCaptures(self, board):
        """
        :type board: List[List[str]]
        :rtype: int
        """
        x, y = -1, -1
        for i in xrange(8):
            for j in xrange(8):
                if board[i][j] == 'R':
                    x, y = i, j
                    break

        res = 0

        for x_step, y_step in [(0, 1), (0, -1), (1, 0), (-1, 0)]:
            i, j = x + x_step, y + y_step
            while i > -1 and i < 8 and j > -1 and j < 8:
                if board[i][j] == 'B':
                    break
                if board[i][j] == 'p':
                    res += 1
                    break
                i, j = i + x_step, j + y_step

        return res

File: 999.Available-Captures-for-Rook.java
// https://leetcode.com/problems/available-captures-for-rook/
//
// algorithms
// Easy (65.93%)
// Total Accepted: 22,438
// Total Submissions: 34,035
// beats 100.0% of java submissions


class Solution {
    public int numRookCaptures(char[][] board) {
        int res = 0;
        int x = -1, y = -1;
        int row = board.length, col = board[0].length;
        boolean rookFound = false;

        for (int i = 0; i < row; i++) {
            for (int j = 0; j < col; j++) {
                if (board[i][j] == 'R') {
                    x = i;
                    y = j;
                    rookFound = true;
                    break;
                }
            }
            if (rookFound) {
                break;
            }
        }

        for (int i = x - 1; i >= 0; i--) {
            if (board[i][y] == 'p') {
                res++;
                break;
            } else if (board[i][y] == 'B') {
                break;
            }
        }

        for (int i = x + 1; i < row; i++) {
            if (board[i][y] == 'p') {
                res++;
                break;
            } else if (board[i][y] == 'B') {
                break;
            }
        }

        for (int i = y - 1; i >= 0; i--) {
            if (board[x][i] == 'p') {
                res++;
                break;
            } else if (board[x][i] == 'B') {
                break;
            }
        }

        for (int i = y + 1; i < col; i++) {
            if (board[x][i] == 'p') {
                res++;
                break;
            } else if (board[x][i] == 'B') {
                break;
            }
        }

        return res;

    }
}